js/maps.js

```diff
@@ -220,13 +209,32 @@ function zoomToFit() {
     });
   map.fitBounds(bounds);
   map.panToBounds(bounds);
-  
+
 }
 
 MAP_SELECT.addEventListener('change', toggleLayer);
 MAP_THRESHOLD.addEventListener('change', renderMap);
 ZOOM_TO_FIT.addEventListener('click', zoomToFit);
 
-API[getLayers](eventName)
+const center = new google.maps.LatLng(15.231458142, -61.2507115);
+const map = new google.maps.Map(MAP_CONTAINER, Object.assign(MAP_OPTIONS, { center }));
+
+const eventName = queryParams().event;
+const pendingLayers = queryParams().pending;
+let getLayerFunc = 'layers';
+let layer = queryParams().layer;
+let zoomFunc = FitEventBounds;
+// pending query param overrides the layer query param
+// layer is used to overide the map to only show
+// a single layer source, e.g. for snapshots
+if (pendingLayers) {
+  getLayerFunc = 'pendingLayers';
+} else if (layer) {
+  getLayerFunc = 'layer';
+  zoomFunc = zoomToFit;
+}
+
+API[getLayerFunc](eventName, layer)
   .then(buildLayersMenu)
-  .then(renderMapAndFit);
+  .then(renderMap)
```

Comment:
I don't think `renderMap` is a promise, so I think the next then block will run immediately.

Reply:
Ignore that, [MDN says it has good support now](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Browser_compatibility).